cmd/controller-manager/app/options/options.go

```diff
@@ -140,7 +142,17 @@ func (o *Options) AddFlags(flags *pflag.FlagSet, allControllers []string, disabl
 		&o.PrometheusQuantiles,
 		"prometheus-quantiles",
 		map[string]string{"0.5": "0.01", "0.95": "0.01", "0.99": "0.002"},
-		"prometheus summary objective quantiles",
+		"Prometheus summary objective quantiles",
+	)
+
+	flags.StringToStringVar(
+		&federatedcluster.VirtualNodeLabels,
+		"virtual-node-labels",
+		map[string]string{
+			"type":                             "virtual-kubelet",
+			"node.kubernetes.io/instance-type": "eklet",
```

Comment:
This feels very vendor-specific. What about moving this to the FederatedCluster definition?

Reply:
@limhawjia suggests adding command line parameters to facilitate user configuration. I also think "node.kubernetes.io/instance-type": "eklet" very vendor-specific, maybe we can delete it.

Reply:
let's keep `"type": "virtual-kubelet"` and remove `"node.kubernetes.io/instance-type": "eklet"`?

Reply:
Let's remove both? vk is *supposed* to be schedulable; it is just inappropriate in the eks case.

Reply:
vk is supposed to be schedulable, but there's another problem that makes us should filter it.
If a cluster contains virtual nodes, federation will perceive a significant amount of available resources in that cluster during resource scheduling. This leads to a disproportionately high scheduling weight for that cluster type. Consequently, during dynamic weight scheduling, most replicas will be scheduled to this cluster.

Reply:
After discussing with @SOF3 , we have concluded that this may not be a general requirement. so i will not add the default parameter here. cc @limhawjia